04_numeronym/CameronSchafer/main.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	var s = make([]string, len(vals))
+	copy(s, vals)
+
+	for i, str := range s {
```

Comment:
IMO it's nicer to factor this assignment into two functions: `numeronyms` and `numeronym`:
```go
func numeronyms(vals ...string) []string {
	result := make([]string, len(vals))
	for i, val := range vals {
                result[i] =  numeronym(val)
        }
        return result
}

func numeronyms(s string) string {
    // ...
}
```

Reply:
Done

Reply:
FWIW i meant including the whole logical test as well, so that numeronym:
```go
func numeronym(r []rune) string {
         r := []rune(str)
	if len(r) <= 3 {
		return s
	}
	return fmt.Sprintf("%c%d%c", r[0], len(r)-2, r[len(r)-1])
}
```

No to worry, next time ;)